checks/apps/gromacs/gromacs_benchpep-h.py

```diff
@@ -0,0 +1,174 @@
+        self.descr = f'GROMACS {self.__bench} GPU benchmark (update mode: {self.update_mode}, bonded: {self.bonded_impl}, non-bonded: {self.nb_impl})'
+        bench_file_path = os.path.join(self.current_system.resourcesdir, 'gromacs-benchmarks', 'www.mpinat.mpg.de', self.__bench, f'{self.__bench}.tpr')
+        self.prerun_cmds = [
+            f'ln -s {bench_file_path} benchmark.tpr'
+        ]
+        if self.fft_variant == 'heffte':
+            npme_ranks = 2*self.num_nodes
+        else:
+            npme_ranks = 1
+        self.executable_opts += [
+            '-nsteps 10000',
+            '-nstlist 400',
+            '-noconfout',
+            '-nb', self.nb_impl,
+            '-npme', f'{npme_ranks}',
+            '-pme', 'gpu', 
+            '-update', self.update_mode,
+            '-bonded', self.bonded_impl,
+            '-pin on',
+            '-s benchmark.tpr'
+        ]
+        self.env_vars = {
+            'MPICH_GPU_SUPPORT_ENABLED': '1',
+            'GMX_ENABLE_DIRECT_GPU_COMM': '1',
+            'GMX_FORCE_GPU_AWARE_MPI': '1',
+            'OMP_NUM_THREADS': '6',
+            'GMX_GPU_PME_DECOMPOSITION': '1',
+            'GMX_PMEONEDD': '1'
+        }
+
+    @run_after('init')
+    def set_modules(self):
+        match self.fft_variant:
+            case 'heffte':
+                self.modules = ['GROMACS/2023.2-cpeGNU-22.12-HeFFTe-GPU']
+            case 'vkfft':
+                self.modules = ['GROMACS/2023.2-cpeGNU-22.12-VkFFT-GPU']
+            case _:
+                self.skip('FFT library not defined')
+
+    @run_before('run')
+    def set_cpu_mask(self):
+        cpu_bind_mask = '0xfe000000000000,0xfe00000000000000,0xfe0000,0xfe000000,0xfe,0xfe00,0xfe00000000,0xfe0000000000'
+        self.job.launcher.options = [f'--cpu-bind=mask_cpu:{cpu_bind_mask}']
+
+    @run_after('init')
+    def add_select_gpu_wrapper(self):
+        self.prerun_cmds += [
+            'cat << EOF > select_gpu',
+            '#!/bin/bash',
+            'export ROCR_VISIBLE_DEVICES=\$SLURM_LOCALID',
+            'exec \$*',
+            'EOF',
+            'chmod +x ./select_gpu'
+        ]
+        self.executable = './select_gpu ' + self.executable
+
+
+    @run_after('init')
+    def setup_nb(self):
+        valid_systems = {
+            'heffte': {
+                1: ['lumi:gpu'],
+                2: ['lumi:gpu'],
+            },
+            'vkfft': {
+                1: ['lumi:gpu'],
+            }
+        }
+        try:
+            self.valid_systems = valid_systems[self.fft_variant][self.num_nodes]
+        except KeyError:
+            self.valid_systems = []
+
+
+    @performance_function('ns/day')
+    def perf(self):
+        return sn.extractsingle(r'Performance:\s+(?P<perf>\S+)',
+                                'md.log', 'perf', float)
+    @deferrable
+    def energy_drift(self):
+        return sn.extractsingle(r'\s+Conserved\s+energy\s+drift\:\s+(\S+)', 'md.log', 1, float)
```

Comment:
This is good, but we *also* want step 0 energy parsed out, i.e. the first match to the regex like this: https://github.com/reframe-hpc/reframe/blob/5986ed553786ae2e1c781dab5549be1042db779b/hpctestlib/sciapps/gromacs/benchmarks.py#L102